### PR TITLE
perf: faster detection of  invalid filenames

### PR DESCRIPTION
**Pull request by ckerr**
- rewrite the reserved-filename detector to be faster
- move the API into tr_torrent_files
- add tests

### CODE REVIEW

File: libtransmission/torrent-files.cc [hunk 1]
```diff
@@ -310,3 +310,114 @@ void tr_torrent_files::remove(std::string_view parent_in, std::string_view tmpdi
...
+        "COM8"sv, "COM9"sv, "LPT1"sv, "LPT2"sv, "LPT3"sv, "LPT4"sv, "LPT5"sv, "LPT6"sv, "LPT7"sv, "LPT8"sv, "LPT9"sv,
+    };
+    if (std::find(std::begin(ReservedNames), std::end(ReservedNames), in_upper_sv) != std::end(ReservedNames))
+    {
+        return true;
+    }
+
+    static auto constexpr ReservedPrefixes = std::array<std::string_view, 22>{
+        "AUX."sv, "CON."sv,  "NUL."sv,  "PRN."sv,  "COM1."sv, "COM2"sv,  "COM3."sv, "COM4."sv, "COM5."sv, "COM6."sv, "COM7."sv,
+        "COM8"sv, "COM9."sv, "LPT1."sv, "LPT2."sv, "LPT3."sv, "LPT4."sv, "LPT5."sv, "LPT6."sv, "LPT7."sv, "LPT8."sv, "LPT9."sv,
```
Comment (ile6695) [inline libtransmission/torrent-files.cc:353]:
```suggestion
        "COM8."sv, "COM9."sv, "LPT1."sv, "LPT2."sv, "LPT3."sv, "LPT4."sv, "LPT5."sv, "LPT6."sv, "LPT7."sv, "LPT8."sv, "LPT9."sv,
```
:)

Comment (s12chung) [inline libtransmission/torrent-files.cc:353]:
```suggestion
        "AUX."sv, "CON."sv,  "NUL."sv,  "PRN."sv,  "COM1."sv, "COM2."sv,  "COM3."sv, "COM4."sv, "COM5."sv, "COM6."sv, "COM7."sv,
        "COM8."sv, "COM9."sv, "LPT1."sv, "LPT2."sv, "LPT3."sv, "LPT4."sv, "LPT5."sv, "LPT6."sv, "LPT7."sv, "LPT8."sv, "LPT9."sv,
```

Randomly reading PRs and following along and saw this after seeing the other comment. No idea what the code does.

Comment (s12chung) [reply libtransmission/torrent-files.cc:353]: Ah... I see. Good random fact to know :)

Comment (ckerr) [reply libtransmission/torrent-files.cc:353]: Thanks!

Comment (ile6695) [reply libtransmission/torrent-files.cc]: @ckerr You might also want to check the other suggested change that I missed. Sorry about that.

Comment (s12chung) [reply libtransmission/torrent-files.cc]: Think @ile6695 is referring to the `COM2.` change I suggested here:

https://github.com/transmission/transmission/pull/3126#discussion_r879729162

Don't think it was merged...

**Pull request merged**